static/backbone/js/api/lib/maps/data/viewLoader.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Created by zmmachar on 12/17/14.
+ */
+define(["models/view",
+        "collections/projects",
+        "jquery",
+        "config"
+    ],
+    function (View, Projects, $, Config) {
+        'use strict';
+
+        /**
+         * The Viewloader class just structes the data
+         * necessary to load a view
+         * @class ViewLoader
+         */
+        var ViewLoader = function (app) {
+            var updateCollection = function (opts) {
+                if (!opts) {
+                    //TODO: create standardized way to report errors.
+                    return;
+                }
+                var key = opts.key,
+                    models = opts.models,
+                    collectionOpts;
+                if (!this.collections[key]) {
+                    collectionOpts = { key: key, name: opts.name };
+                    //A few special hacks for form data:
+                    if (key.indexOf("form") !== -1) {
+                        collectionOpts.url = '/api/0/forms/' + key.split("_")[1] + '/data/';
+                    }
+                    this.collections[key] = new opts.Collection([], collectionOpts);
+
+                    // important: this trigger enables the overlayManager
+                    // to create a new overlay for each model where the
+                    // GeoJSON geometry is defined.
+                    this.app.vent.trigger('new-collection-created', {collection: this.collections[key],
+                                                                     startVisible: true});
+
+                }
+                this.collections[key].add(models, {merge: true});
+            };
+            /**
+             * A dictionary of the various data types available (given
+             * the projects that have been selected), and the corresponding
+             * data records that have been pulled down from the Data API.
+             */
+            this.collections = {};
+
+            this.initialize = function (opts) {
+                this.view = new View(opts.view);
+                this.app = opts.app;
+                var that = this;
+                this.app.vent.once('map-ready', function () {
```

Comment:
We can move the initialization to the basemap I guess if you'd like (though the application should be able to indicate it wants changes to happen to the map in ways other than directly accessing the basemap object, in general), but the underlying issue is just that there isn't another good way I can think of to indicate that the map is 'able to be manipulated'.  I'm trying to avoid a race condition whereby the view loader (or, whatever other application) ends up executing in the middle of what can be a somewhat lengthy initialization process.  Looking at it, I suppose this establishes a race in the other direction.  This is also the issue with the 'changeZoom' method, which should actually be pulled out of that idle listener.